sql/core/src/main/scala/org/apache/spark/sql/Column.scala

```diff
@@ -822,7 +822,7 @@ class Column(protected[sql] val expr: Expression) extends Logging {
    * @group expr_ops
    * @since 1.3.0
    */
-  def cast(to: String): Column = cast(DataTypeParser.parse(to))
```

Comment:
actually i just realized DataTypeParser.parse already does what DataType.fromString is doing. Maybe we can use that instead and remove DataType.fromString? Sorry for noticing this only now.